src/gf.c

```diff
@@ -126,7 +126,7 @@ const struct jl_typemap_info method_defs = {
     0, &jl_method_type
 };
 const struct jl_typemap_info lambda_cache = {
-    0, &jl_method_instance_type
+    1, &jl_method_instance_type
```

Comment:
I believe such a flag would enable the optimization I'm attempting here. It's only safe to split on the second argument of `(::AbstractThing, ::Int)` if we know `(::AbstractThing{N}, ::Vararg{Int,N})` is not going to exist.